Canvas grid visuals, scrolling, and bottom bar

tlein opened the pull request:
Resolves #31 
Resolves #33

tlein committed:
implement grid

tlein committed:
implement scale

tlein committed:
implement bottom bar

tlein committed:
remove label from vector component and create a distinct label component

tlein committed:
add 'Show Grid?' checkbox to bottom canvas bar

tlein committed:
add map scroll tool

jeffswenson commented:
You should pull in iconic as an npm module instead of embedding it in the source.

Tucker: done

tlein committed:
use open-iconic npm package instead of including it

tlein committed:
add shadow effect to canvas and change background color

tlein committed:
implement scrolling with the canvas

tlein committed:
change splash screen icon to use font icons

tlein committed:
draw canvas lines the same width regardless of scale

tlein committed:
move map-editor logic out of canvas

jeffswenson commented on src/duckling/canvas/canvas.component.ts:
```diff
@@ -6,92 +6,210 @@ import {
+
+    ngAfterViewInit() {
+        this._viewInited = true;
+        this._window.on('resize', () => this.onResize());
```
This handler needs to be removed in ngOnDestroy

jeffswenson commented on src/duckling/canvas/drawing/util.ts:
```diff
@@ -32,3 +32,29 @@ export function drawX(centerPosition : Vector, dimension : Vector, graphics : Gr
 }
+
+/**
+ * Draws a grid at the given position.
```
Remove the types from this comment.

jeffswenson commented on src/duckling/canvas/map-editor.component.ts:
```diff
@@ -1,63 +1,204 @@
+
+    /**
+     * Size of the stage being edited
+     * @type {Vector}
```
Remove all of these unused @type annotations.

tlein committed:
remove content editable from canvas for copy/paste to work with the virtual canvas

tlein committed:
fix issues found in review

tlein committed:
abstract out the canvas drawn elements from map-editor

tlein committed:
remove draggedElement from map move tool and insted use the canvas instance now on the CanvasMouseEvent

tlein committed:
Wrapped CanvasMouseEvent creation in a function
Clear the resize handler in canvas
Change CanvasMouseEvent to an interface
Remove unnecessary setTimeout in splash component
